### PR TITLE
Allowing reordering of people in AddPeopleActivity

### PR DESCRIPTION
**Pull request by shahmharsh**
- Added a ItemTouchHelper which listens to callback and notfies the adapter
- Also fixed bug where 'generate code' button was showing above keyboard
- Also made sure scroll shadow is never shown in 'add_person_recycler_view'

@raincoatforever

### CODE REVIEW

File: app/src/main/java/co/minesweepers/birthday/adapters/AddPeopleAdapter.java [hunk 1]
```diff
@@ -49,17 +54,59 @@ public void onBindViewHolder(ViewHolder holder, int position) {
...
+            for (int i = fromPosition; i < toPosition; i++) {
+                Collections.swap(mPeople, i, i + 1);
+            }
+        } else {
+            for (int i = fromPosition; i > toPosition; i--) {
+                Collections.swap(mPeople, i, i - 1);
+            }
+        }
+        notifyItemMoved(fromPosition, toPosition);
+        return true;
```
Comment (raincoatforever) [inline app/src/main/java/co/minesweepers/birthday/adapters/AddPeopleAdapter.java]: why return boolean if its always true ?

File: app/src/main/java/co/minesweepers/birthday/adapters/helpers/ItemTouchHelperCallback.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    private ItemTouchEventsListener mListener;
+
+    public ItemTouchHelperCallback(@NonNull ItemTouchEventsListener listener) {
+        mListener = listener;
+    }
+
+    @Override
+    public int getMovementFlags(RecyclerView recyclerView, RecyclerView.ViewHolder viewHolder) {
+        int dragFlags = ItemTouchHelper.UP | ItemTouchHelper.DOWN;
+        return makeMovementFlags(dragFlags, 0); // TODO: allow swipe movement for deleting
```
Comment (raincoatforever) [inline app/src/main/java/co/minesweepers/birthday/adapters/helpers/ItemTouchHelperCallback.java:18]: swipe to delete seems very iOS like action .  In most places on android, standard seems to be long press and then a delete button shows up . Lets check for a few examples before deciding to implement that one .

**Comment by raincoatforever** [comment]
Code wise looks good . Maybe we should take another go at UX to see if the use case becomes more obvious ?
Also  , remember discussing that instead of inline editing a person's name , lets go to CreateMemoryActivity to allow that . That could change some stuff in this impl .Maybe just a task to track that is fine for now .

File: app/src/main/java/co/minesweepers/birthday/adapters/helpers/ItemTouchHelperCallback.java (hunk 2 above)
Comment (shahmharsh) [reply app/src/main/java/co/minesweepers/birthday/adapters/helpers/ItemTouchHelperCallback.java:18]: Yes thats why haven't implemented the delete yet

File: app/src/main/java/co/minesweepers/birthday/adapters/AddPeopleAdapter.java (hunk 1 above)
Comment (shahmharsh) [reply app/src/main/java/co/minesweepers/birthday/adapters/AddPeopleAdapter.java]: Changed it to void

**Comment by shahmharsh** [comment]
Comment addressed.
Task added: https://github.com/minesweepers-co/birthday/projects/1#card-2079923
For the UX, I'll talk to Nidhi to see if she can give more options for reordering use case

**Comment by raincoatforever** [comment]
:+1:

**Pull request merged**